recipes/json-schema-validator/all/conanfile.py

```diff
@@ -0,0 +1,75 @@
+import os
+import glob
+from conans import ConanFile, CMake, tools
+from conans.tools import Version
+
+
+class JsonSchemaValidatorConan(ConanFile):
+    name = "json-schema-validator"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/pboettch/json-schema-validator"
+    description = "JSON schema validator for JSON for Modern C++ "
+    topics = ("json-schema-validator", "modern-json",
+              "nlohmann_json", "conan-recipe")
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "cmake"
+    exports_sources = ["CMakeLists.txt"]
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    short_paths = True
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        compiler_version = Version(self.settings.compiler.version)
+        if self.settings.os == "Windows"and self.settings.compiler == "Visual Studio" and compiler_version < "16":
+            tools.check_min_cppstd(self, "17")
```

Comment:
why does it require C++17 only for Windows? Also, you only should use `check_min_cppstd` when cppstd is declared and configured, which is not the default behavior, thus, check it first, otherwise, you can check the compiler version instead.

Reply:
there's a bug in upstream https://github.com/pboettch/json-schema-validator/issues/106

I had done that here
https://github.com/conan-io/conan-center-index/blob/759606922e8766cc1293bd39985ebe7d5a55526d/recipes/json-schema-validator/all/conanfile.py#L41-L42

But how do I indicate that it is required, it needs to be configured in order for the project to compile?

Reply:
That's weird, because `attributes` have been supported since Visual Studio 2015: https://en.cppreference.com/w/cpp/compiler_support

Reply:
Yes but the ` [[maybe_unused]]` is a c++ 17 feature and 2017 is 14 by default so it needs to be explicitly set